src/process-events.c

```diff
@@ -448,59 +439,73 @@ static __always_inline int write_path(struct pt_regs *ctx, dentry_pair *dentries
 
     offset = get_offset(CRC_DENTRY_D_PARENT);
     if (offset == NULL) return -1;
-    u32 parent = *(u32 *)offset; // offset of d_parent
+    u32 dentry_parent = *(u32 *)offset; // offset of d_parent
+
+    offset = get_offset(CRC_MOUNT_MNTPARENT);
+    if (offset == NULL) return -1;
+    u32 mnt_parent_offset = *(u32 *)offset; // offset of mount->mnt_parent
+
+    offset = get_offset(CRC_VFSMOUNT_MNTROOT);
+    if (offset == NULL) return -1;
+    u32 mnt_root_offset = *(u32 *)offset; // offset of vfsmount->mnt_root
+
+    offset = get_offset(CRC_MOUNT_MOUNTPOINT);
+    if (offset == NULL) return -1;
+    u32 mountpoint_offset = *(u32 *)offset; // offset of mount->mountpoint
+
+    offset = get_offset(CRC_MOUNT_MNT);
+    if (offset == NULL) return -1;
+    u32 mnt_offset = *(u32 *)offset; // offset of mount->mnt
```

Comment:
all of these offsets are retrieved in this way rather than our helper methods because we want to "cache" the offset values instead of getting them from the map on every iteration of the loop

Reply:
It's kind of gross, but until we can do load-time relocations it's probably a reasonable tradeoff.